gym_hil/envs/panda_arrange_boxes_gym_env.py

```diff
@@ -0,0 +1,186 @@
+        self.apply_action(action)
+
+        # Compute observation, reward and termination
+        obs = self._compute_observation()
+        rew = self._compute_reward()
+        success = self._is_success()
+
+        if self.reward_type == "sparse":
+            success = rew == 1.0
+
+        # Check if block is outside bounds
+        block_pos = self._data.sensor("block1_pos").data
+        exceeded_bounds = np.any(block_pos[:2] < (_SAMPLING_BOUNDS[0] - self.block_range - 0.05)) or np.any(
+            block_pos[:2] > (_SAMPLING_BOUNDS[1] + self.block_range + 0.05)
+        )
+
+        terminated = bool(success or exceeded_bounds)
+
+        return obs, rew, terminated, False, {"succeed": success}
+
+    def _compute_observation(self) -> dict:
+        """Compute the current observation."""
+        # Create the dictionary structure that matches our observation space
+        observation = {}
+
+        # Get robot state
+        robot_state = self.get_robot_state().astype(np.float32)
+
+        # Assemble observation respecting the newly defined observation_space
+        block_pos = self._data.sensor("block1_pos").data.astype(np.float32)
+
+        if self.image_obs:
+            # Image observations
+            front_view, wrist_view = self.render()
+            observation = {
+                "pixels": {"front": front_view, "wrist": wrist_view},
+                "agent_pos": robot_state,
+            }
+        else:
+            # State-only observations
+            observation = {
+                "agent_pos": robot_state,
+                "environment_state": block_pos,
+            }
+
+        return observation
+
+    def _get_sensors(self):
+        block_sensors = [self._data.sensor(f"block{i}_pos") for i in range(1,self.no_blocks+1)]
+        target_sensors = [self._data.sensor(f"target{i}_pos") for i in range(1,self.no_blocks+1)]
+        return block_sensors, target_sensors
+    
+    def _compute_reward(self) -> float:
+        block_sensors, target_sensors = self._get_sensors()
+        block_target_pairs = zip(block_sensors, target_sensors)
+
+        if self.reward_type == "dense":
+            r_close = list(map(lambda pair: np.exp(-20 * np.linalg.norm(pair[0].data-pair[1].data)), block_target_pairs))
+            return sum(r_close)
+        else:
+            r_lift = list(map(lambda block: 0.3 * float(block.data[2] < 0.1), block_sensors))
+            r_close = list(map(lambda pair: 0.7 * np.exp(-20 * np.linalg.norm(pair[0].data-pair[1].data)), block_target_pairs))
+            return sum(r_lift) + sum(r_close)
+    
+    def _is_success(self) -> bool:
+        block_sensors, target_sensors = self._get_sensors()
+        block_target_pairs = zip(block_sensors, target_sensors)
+        distances = list(map(lambda pair: np.linalg.norm(pair[0].data-pair[1].data), block_target_pairs))
+        return all(list(map(lambda dist: dist < 0.1, distances))) and all(list(map(lambda block: block.data[2] < 0.1)))
```

Comment:
Could you remove check for lift, we don't need it in this task. And decrease the threshold distance to like 0.03?
```suggestion
        return all(list(map(lambda dist: dist < 0.03, distances)))
```